allennlp_models/generation/dataset_readers/cnn_dm.py

```diff
@@ -0,0 +1,165 @@
+from allennlp.data.instance import Instance
+from allennlp.data.tokenizers import Tokenizer, SpacyTokenizer
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+
+logger = logging.getLogger(__name__)
+
+
+@DatasetReader.register("cnn_dm")
+class CNNDailyMailDatasetReader(DatasetReader):
+    """
+    Read a tsv file containing paired sequences, and create a dataset suitable for a
+    `ComposedSeq2Seq` model, or any model with a matching API.
+
+    Expected format for each input line: <source_sequence_string>\t<target_sequence_string>
+
+    The output of `read` is a list of `Instance` s with the fields:
+        source_tokens : `TextField` and
+        target_tokens : `TextField`
+
+    `START_SYMBOL` and `END_SYMBOL` tokens are added to the source and target sequences.
+
+    # Parameters
+
+    source_tokenizer : `Tokenizer`, optional
+        Tokenizer to use to split the input sequences into words or other kinds of tokens. Defaults
+        to `SpacyTokenizer()`.
+    target_tokenizer : `Tokenizer`, optional
+        Tokenizer to use to split the output sequences (during training) into words or other kinds
+        of tokens. Defaults to `source_tokenizer`.
+    source_token_indexers : `Dict[str, TokenIndexer]`, optional
+        Indexers used to define input (source side) token representations. Defaults to
+        `{"tokens": SingleIdTokenIndexer()}`.
+    target_token_indexers : `Dict[str, TokenIndexer]`, optional
+        Indexers used to define output (target side) token representations. Defaults to
+        `source_token_indexers`.
+    source_add_start_token : bool, (optional, default=True)
+        Whether or not to add `START_SYMBOL` to the beginning of the source sequence.
+    source_add_end_token : bool, (optional, default=True)
+        Whether or not to add `END_SYMBOL` to the end of the source sequence.
+    """
+
+    def __init__(
+        self,
+        source_tokenizer: Tokenizer = None,
+        target_tokenizer: Tokenizer = None,
+        source_token_indexers: Dict[str, TokenIndexer] = None,
+        target_token_indexers: Dict[str, TokenIndexer] = None,
+        source_max_tokens: Optional[int] = None,
+        target_max_tokens: Optional[int] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self._source_tokenizer = source_tokenizer or SpacyTokenizer()
+        self._target_tokenizer = target_tokenizer or self._source_tokenizer
+        self._source_token_indexers = source_token_indexers or {"tokens": SingleIdTokenIndexer()}
+        self._target_token_indexers = target_token_indexers or self._source_token_indexers
+        self._source_max_tokens = source_max_tokens
+        self._target_max_tokens = target_max_tokens
+
+    @staticmethod
+    def _hashhex(url):
+        h = hashlib.sha1()
+        h.update(url)
+        return h.hexdigest()
+
+    @staticmethod
+    def _sanitize_story_line(line):
+        line = ftfy.fix_encoding(line)
```

Comment:
I think there are more aggressive functions in `ftfy` you could try here, but they all make it slower.